File: app/src/main/java/com/mytaxiapp/di/module/RetrofitModule.java
package com.mytaxiapp.di.module;


import com.mytaxiapp.data.remote.VehicleInterface;

import dagger.Module;
import dagger.Provides;
import retrofit.RestAdapter;


/**
 * @author Tosin Onikute.
 */

@Module
public class RetrofitModule {

    @Provides
    public VehicleInterface providesVehicleInterface(RestAdapter restAdapter) {
        return restAdapter.create(VehicleInterface.class);
    }
}

File: app/src/main/java/com/mytaxiapp/di/module/RetrofitModule.kt
package com.mytaxiapp.di.module


import com.mytaxiapp.data.remote.VehicleInterface

import dagger.Module
import dagger.Provides
import retrofit.RestAdapter


/**
 * @author Tosin Onikute.
 */

@Module
class RetrofitModule {

    @Provides
    fun providesVehicleInterface(restAdapter: RestAdapter): VehicleInterface {
        return restAdapter.create(VehicleInterface::class.java)
    }
}